temporian/test/api_test.py

```diff
@@ -55,9 +48,31 @@ def disabled_test_create_processor(self):
                 a: input_signal_data,
             },
         )
-
         logging.info("results: %s", results)
 
+    def test_serialization(self):
+        a = t.place_holder(
+            features=[
+                t.Feature(name="f1", dtype=t.dtype.FLOAT),
+                t.Feature(name="f2", dtype=t.dtype.FLOAT),
+            ],
+            index=[],
+        )
+        b = t.sma(data=a, window_length=7)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            path = os.path.join(tempdir, "my_processor.tem")
+            t.save(
+                inputs={"a": a},
```

Comment:
Gave some thought to what we discussed today about needing to specify inputs 💭 

We think this could be optional if the user wants to specify some intermediate events as a cutoff for creating the graph, but if not default we should default to creating the graph from the `outputs` all the way up to events that don't have a `creator` (raw input events). Same goes for the `inputs` param in the `infer_processor` method - by default we should create and save the whole graph, crawling up from the `output` ones until we reach events with no creator (similar as the evaluator's logic)

See related comment about the place holder op not being strictly necessary if we treat events without a creator as input events implicitly.

Let's dedicate some time to this in tuesday's meeting if necessary!

Reply:
I like the idea of optional simplification. Let's discuss tuesday about the logic we want.

Question: Imagine a situation of a graph with two input events. How is the user specifying the input of each specific event? 

Bonus: I am wondering if the use of the processor directly would be a good idea. Here is how it would look like.

p = t.infer_processor(....)
result = p(event_1=..., event_2=...) # Equivalent to t.evaluate(inputs={p.inputs()["event_1"]:..., p.inputs()["event_2"]:...}, outputs=p.outputs()
p.save(path=...)
p2 = t.load(path=...)

Reply:
> Imagine a situation of a graph with two input events. How is the user specifying the input of each specific event?

Currently in `evaluate` the user specifies the input events [using an input `Event`'s ref](https://github.com/google/temporian/blob/main/temporian/test/prototype_test.py#L77), but we might need to change that to using the event's unique name or id since we won't have that ref after serializing and deserializing.

I think this is independent from the serialization and deserialization logic though, which should just walk back from the specified output event until source events (i.e., events with no incoming edges in the graph, i.e. events without a creator, i.e. "input events") unless the user specifies otherwise with your proposed syntax.